### ISSUE TITLE
Swagger api not showing

### ISSUE THREAD

**Issue by brattolo** [RSuter/NSwag#1628]
![image](https://user-images.githubusercontent.com/36637783/45931350-57ca0d00-bf6d-11e8-9cf2-5648c1fa3da9.png)

I have a issue where i go to /swagger the explorer field at the top is just empty. It could be intended behavior. However, when viewing documentation/samples it appears to always be set to the _swagger.json_ of the api its configured on. I want to make _swagger/v1/swagger.json_ the default explorer value

When i enter the swagger.json url manually in the explorer it does work.
I tried adding to the config.SwaggerRoutes manually but this didn't work.
Is there a different configuration property i am missing to set this manually?
Is there something causing this?

Thanks in advance.

```
        public IConfiguration Configuration { get; }

        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }


        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services
                .Configure<Configuration>(x => this.Configuration.Bind(x));

            var serviceProvider = services.BuildServiceProvider();
            var configuration = serviceProvider.GetRequiredService<IOptions<Configuration>>();

            services
                .AddAuthentication(IdentityServerAuthenticationDefaults.AuthenticationScheme)
                .AddIdentityServerAuthentication(options => {
                    // auth server base endpoint (will use to search for disco doc)
                    options.Authority = configuration.Value.ASPNETCORE_AUTH_URLS_HTTPS;
                    options.ApiName = "demo_api"; // required audience of access tokens
                    options.RequireHttpsMetadata = true;
                });

            services
                .AddMvc()
                .SetCompatibilityVersion(CompatibilityVersion.Version_2_2);

            services.AddSwagger();
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IHostingEnvironment env, IOptions<Configuration> configuration)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            else
            {
                // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
                app.UseHsts();
            }

            app.UseHttpsRedirection();

            app.UseStaticFiles();

            app.UseAuthentication();

            app.UseSwaggerUi3WithApiExplorer(settings =>
            {
                settings.GeneratorSettings.DefaultPropertyNameHandling = PropertyNameHandling.CamelCase;
                settings.GeneratorSettings.OperationProcessors.Add(new OperationSecurityScopeProcessor("oauth2"));
                settings.GeneratorSettings.DocumentProcessors.Add(
                    new SecurityDefinitionAppender("oauth2", new SwaggerSecurityScheme {
                        Type = SwaggerSecuritySchemeType.OAuth2,
                        Flow = SwaggerOAuth2Flow.Implicit,
                        AuthorizationUrl = $"{configuration.Value.ASPNETCORE_AUTH_URLS_HTTPS}/connect/authorize",
                        Scopes = new Dictionary<string, string> { { "demo_api", "Demo API - full access" } }
                 }));
                settings.OAuth2Client = new OAuth2ClientSettings {
                    ClientId = "demo_api_swagger",
                    AppName = "Demo API - Swagger"
                };
            });

            app.UseMvcWithDefaultRoute();
        }

```

**Comment by akamyshanov** [comment]
So, what is the solution? Also struggling with the default URL in the Swagger UI.

NSwag.AspNetCore v11.20.1

**Comment by brattolo** [comment]
I created a new sample with only nswag configured and it worked. Figured it had something to do with IIS profile, because i also made changes to my launch profile and the new project had the default settings. However did not figure out exactly what the issue was. I also could not use redirects for some strange reason. Hopefully this might help you a little in the right direction


Met vriendelijke groeten,
Bart Gevaert
